src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/Constants.cs

```diff
@@ -78,5 +78,13 @@ internal static class Constants
             typeof(T) == typeof(UInt128) ||
 #endif
             typeof(T).IsEnum;
+
+        // for these types GetHashCode returns their value casted to int, so when we receive a Dictionary/HashSet where there are key
+        // we know that all hash codes are unique and we can avoid some work later
+        internal static bool KeysAreHashCodes<T>()
+            => typeof(T) == typeof(int) || typeof(T) == typeof(uint)
+            || typeof(T) == typeof(short) || typeof(T) == typeof(ushort)
+            || typeof(T) == typeof(byte) || typeof(T) == typeof(sbyte)
+            || ((typeof(T) == typeof(IntPtr) || typeof(T) == typeof(UIntPtr)) && IntPtr.Size == 4);
```

Comment:
nit: nint and nuint?

Reply:
These are already handled by `{U]IntPtr`. See [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWAChis8BOACgDEB7WgHgEkYEAFBZAPkoEpeA3KQo16DAKot2nHvyFEyVOoxiwEswcKViYqNRvnCAbAAJg9ADYnlDACqziAb2InXJrAHYTCAJ4AHAFNaADNKW14TAF5I738g0KkOZAiMDFjAkLCI6PT4yklWJM0iAF8gA=) for demo.

Reply:
No, I meant weet should write "nint" and "nuint" instead of "IntPtr" abd "UIntPtr". It was a style comment.